Detectors/ITSMFT/ITS/tracking/src/IOUtils.cxx

```diff
@@ -107,8 +107,12 @@ void ioutils::loadEventData(ROframe& event, gsl::span<const itsmft::CompClusterE
   int clusterId{0};
 
   for (auto& c : clusters) {
-    int layer = geom->getLayer(c.getSensorID());
+    if (geom->getHalfBarrel(c.getSensorID()) == ignBarrel) { // Top barrel Id: 0, Bot: 1, never ignored otherwise
```

Comment:
`getHalfBarrel` does some looping and calculation, better first check if the selection is requested at all:
```
if (ignBarrel<2 && geom->getHalfBarrel(c.getSensorID())) {
```